tests/jobs/localisation/example_case/scripts/common_functions.py

```diff
@@ -0,0 +1,604 @@
+    seed_file: str = "randomseeds.txt"
+    variogram: str = "exponential"
+    correlation_range: Tuple[float] = (3000.0, 1000.0, 2.0)
+    correlation_azimuth: float = 45.0
+    correlation_dip: float = 0.0
+    correlation_exponent: float = 1.9
+    trend_use: bool = False
+    trend_params: Tuple[float] = (1.0, -1.0)
+    trend_relstd: float = 0.05
+    grid_dimension: Tuple[int] = (150, 250, 1)
+    grid_file_name: str = "GRID.EGRID"
+
+
+@dataclass
+class Response:
+    """
+    Specify the coarse grid dimensions, name of file and type
+    of average operation to calculated upscaled values that
+    are predictions of observations of the same grid cells.
+    Which cell indices are observed are specified in
+    observation settings.
+    """
+
+    grid_dimension: Tuple[int] = (15, 25, 1)
+    upscaled_file_name: str = "Upscaled.roff"
+    grid_file_name: str = "UpscaleGrid.EGRID"
+    response_function: str = "average"
+    gen_data_file_name: str = "UpscaledField_0.txt"
+    calculate_all_cells: bool = True
+
+
+@dataclass
+class Observation:
+    """
+    Specify name of files for generated observations
+    and also which grid cells from coarse grid is used
+    as observables. (Cells that have values that are used as observations)
+    """
+
+    directory: str = "observations"
+    file_name: str = "observations.obs"
+    data_dir: str = "obs_data"
+    param_file_name: str = "init_files/UpscaledObsField.grdecl"
+    rel_error: float = 0.10
+    min_abs_error: float = 0.01
+    # selected_grid_cells: Tuple[Tuple[int]] = ((5, 10, 1), (10, 5, 1))
+    selected_grid_cells: Tuple[Tuple[int]] = (8, 12, 1)
+
+
+@dataclass
+class Localisation:
+    """
+    Specify settings for the localisation config file.
+    """
+
+    method: str = "gaussian"
+    # method: str = "constant"
+
+
+@dataclass
+class Optional:
+    """
+    Specify if some optional files should be
+    written or not (for QC purpose).
+    """
+
+    write_upscaled_to_file: bool = True
+    write_obs_pred_diff_field_file: bool = True
+
+
+@dataclass
+class Settings:
+    """
+    Settings for the test case
+    """
+
+    grid_size: GridSize = GridSize()
+    field: Field = Field()
+    response: Response = Response()
+    observation: Observation = Observation()
+    localisation: Localisation = Localisation()
+    optional: Optional = Optional()
+
+
+settings = Settings()
+
+
+def generate_field_and_upscale(real_number, grid):
+    seed_file_name = settings.field.seed_file
+    relative_std = settings.field.trend_relstd
+    use_trend = settings.field.trend_use
+    algorithm_method = settings.field.algorithm
+    start_seed = get_seed(seed_file_name, real_number)
+    if algorithm_method == "gstools":
+        print(f"Use algorithm: {algorithm_method}")
+        residual_field = simulate_field_using_gstools(start_seed)
+    else:
+        print("Use algorithm: gaussianfft")
+        residual_field = simulate_field(start_seed)
+    if use_trend == 1:
```

Comment:
I think this should be `if use_trend` since `use_trend` is a `bool`.

Reply:
Fixed